api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -140,6 +142,30 @@ public StructType partitionType() {
     return lazyPartitionType;
   }
 
+  /**
+   * Returns a struct with TransformID's which match the ID's used in the Table Metadata. This is
+   * different than the {@link #partitionType()} method which returns a struct which is guaranteed
+   * not to overlap with column ID's of the table by reassigning ID's.
+   */
+  public StructType rawPartitionType() {
+    if (schema.idsToOriginal().isEmpty()) {
```

Comment:
Is this just the "unpartitioned" case? If so can we condition this on the schema being empty?

If the case is just idToOriginals.isEmpty, shouldn't that be an error? Should we always have values there?

Reply:
Ok my only big question on the PR is do we really need this? Feels like rawPartitionType() is never really important to use right?

Reply:
Yea this is only for the case that ids are not assigned.

Yea one of the main problems was that the partition field ids are persisted on file, and so are needed to deserialize the manifest.  So this is used in ManifestReader in that one case.  Afterwards, everything in memory is re-assigned to de-conflict.

Reply:
I think this is the detail that belong in the java doc. Returns a struct matching partition information as written into manifest files.

Reply:
Good point, redid the javadoc to just explain this, let me know.